modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java

```diff
@@ -65,12 +67,12 @@
     @Inject
     private MuleMetadataManager metadataManager;
 
-    protected ExtensionComponent(RuntimeExtensionModel extensionModel, RuntimeComponentModel componentModel, String configurationProviderName, ExtensionManagerAdapter extensionManager)
+    protected ExtensionComponent(RuntimeExtensionModel extensionModel, String configurationProviderName, ExtensionManagerAdapter extensionManager, MetadataMediator metadataMediator)
     {
         this.extensionModel = extensionModel;
         this.configurationProviderName = configurationProviderName;
         this.extensionManager = extensionManager;
-        this.metadataMediator = new MetadataMediator(extensionModel, componentModel);
+        this.metadataMediator = metadataMediator;
```

Comment:
Actually I was about to make an interface first and them added like this as constructor parameter in order to be able to mock it because when I wanted to test the ExtensionComponent classloader changes I was not able to do it without mocking this class.

What do you think? Should we expose this mediator object to be injected so it can be "changed" later (on this case, mocked)